src/include/txn.h

```diff
@@ -172,18 +172,31 @@ typedef enum __wt_txn_isolation {
  *	records during commit or undo the operations during rollback.
  */
 struct __wt_txn_op {
-	uint32_t fileid;
+	WT_DATA_HANDLE *dhandle;
 	enum {
 		WT_TXN_OP_NONE,
 		WT_TXN_OP_BASIC,
```

Comment:
Is WT_TXN_OP_BASIC for column store that we have a separate one for row store. Probably we can rename it to WT_TXN_OP_BASIC_COL. Similarly for WT_TXN_OP_INMEM to WT_TXN_OP_INMEM_COL.

Reply:
WT_TXN_OP_BASIC represents both column store and row store in case of non-timestamp builds. Introduced WT_TXN_OP_BASIC_ROW and WT_TXN_OP_INMEM_ROW just to call __wt_buf_free in __wt_txn_op_free()

Reply:
I think this would be clearer if the types were always distinguished - does that make this change a lot more invasive?